contracts/allowlist/ValidatorSmartContractAllowList.sol

```diff
@@ -0,0 +1,127 @@
+        require(allowedAccounts[msg.sender].activeValidator, "sender does not have an active validator");
+        allowedAccounts[msg.sender].activeValidator = false;
+        validators[allowedAccounts[msg.sender].validatorIndex] = validators[validators.length-1];
+        validators.pop();
+    }
+
+    function voteToAddAccountToAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == false, "account to add is already on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to add account");
+        }
+        currentVotes[account].push(msg.sender);
+    }
+
+    function voteToRemoveAccountFromAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == true, "account to remove is not on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to remove account");
+        }
+        currentVotes[account].push(msg.sender);
+    }
+
+    function removeVoteForAccount(address account) external senderIsAllowed {
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            if (currentVotes[account][i] == msg.sender) {
+                currentVotes[account][i] = currentVotes[account][currentVotes[account].length-1];
+                currentVotes[account].pop();
+                break;
+            }
+        }
+    }
+
+    function countVotes(address account) external senderIsAllowed returns(uint) {
+        uint numVotes;
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            if (allowedAccounts[currentVotes[account][i]].allowed) {
+                // only increment numVotes if account that voted is still allowed
+                numVotes++;
+            }
+        }
+        if (numVotes > numAllowedAccounts / 2) {
+            delete(currentVotes[account]); // Is it a good idea to delete the array? What else could I do? Array should be max 128 long
+            if (allowedAccounts[account].allowed) {
+                numAllowedAccounts--;
+                if(allowedAccounts[account].activeValidator) {
+                    for(uint i=allowedAccounts[account].validatorIndex; i < validators.length - 1; i++) {
+                        validators[i] = validators[i+1];
+                    }
+                    validators.pop();
+                }
+                delete(allowedAccounts[account]);
+            } else {
+                numAllowedAccounts++;
+                allowedAccounts[account] = accountInfo(true, false, 0);
+            }
+        }
+        return numVotes;
```

Comment:
I am not sure that the number of votes is the most interesting thing to return. Isn't it rather whether the vote has been called or not? Most useful would probably be something like `(numVotes, votesRequired, didTheElectionSucceed)`.